### PR TITLE
VideoPlayer: drop recording related methods

### PR DESCRIPTION
**Pull request by FernetMenta**
@ksooo instant recording is PVR only and shold be handled by PVR directy. Can you help with the pvr code?

### CODE REVIEW

**Comment by ksooo** [comment]
>  Can you help with the pvr code?

Sure. Busy atm, but will do asap.

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -497,8 +487,6 @@ const infomap player_labels[] =  {{ "hasmedia",         PLAYER_HAS_MEDIA },
                                   { "forwarding8x",     PLAYER_FORWARDING_8x },
                                   { "forwarding16x",    PLAYER_FORWARDING_16x },
                                   { "forwarding32x",    PLAYER_FORWARDING_32x },
-                                  { "canrecord",        PLAYER_CAN_RECORD },
-                                  { "recording",        PLAYER_RECORDING },
```
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: The defines for `PLAYER_CAN_RECORD` and `PLAYER_RECORDING` (https://github.com/xbmc/xbmc/blob/master/xbmc/guiinfo/GUIInfoLabels.h#L44) should be removed as well.

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp [hunk 2]
```diff
@@ -243,23 +243,6 @@ CDVDInputStream::ENextStream CDVDInputStreamPVRManager::NextStream()
   return NEXTSTREAM_NONE;
 }
 
-bool CDVDInputStreamPVRManager::CanRecord()
-{
-  if (!m_isRecording)
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: This member can be removed, right?

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp [hunk 3]
```diff
@@ -243,11 +243,6 @@ static int PlayerControl(const std::vector<std::string>& params)
     if( g_application.m_pPlayer->IsPlaying() )
       g_application.m_pPlayer->OnAction(CAction(ACTION_SHOW_VIDEOMENU));
   }
-  else if (paramlow == "record")
-  {
-    if( g_application.m_pPlayer->IsPlaying() && g_application.m_pPlayer->CanRecord())
-      g_application.m_pPlayer->Record(!g_application.m_pPlayer->IsRecording());
-  }
```
Comment (ksooo) [inline xbmc/interfaces/builtins/PlayerBuiltins.cpp]: Isn't this an incompatible JSON-RPC API change as you are removing functionality from the API?

**Review by ksooo — COMMENTED** [review]
Some of the non-videoplayer code you removed most probably should not go away completely, but rewritten to use the respective PVR component functionality. Exemple: CGUIWindowMusicBase changes.  I will take care of that.

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (FernetMenta) [reply xbmc/GUIInfoManager.cpp]: ok

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp (hunk 2 above)
Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: nope, m_isRecoding indicates if a recordings is playing and is still in use

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp (hunk 3 above)
Comment (FernetMenta) [reply xbmc/interfaces/builtins/PlayerBuiltins.cpp]: player built-ins are for use in skins, i.e. <onclick>PlayerControl(Rewind)</onclick>
JSON-RPC is not supposed to use built-ins

Comment (ksooo) [reply xbmc/interfaces/builtins/PlayerBuiltins.cpp]: Okay, but as PlayerControl(Record) is used by skins (checked with Estuary), this is nothing to to get removed, but to get implemented different (PVR).

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: okay, then

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp (hunk 3 above)
Comment (FernetMenta) [reply xbmc/interfaces/builtins/PlayerBuiltins.cpp]: Record is not a player command, hence it got to be removed here. Either we need some other builtin or observe clicks on REC_BTN in the code.

Comment (ksooo) [reply xbmc/interfaces/builtins/PlayerBuiltins.cpp]: Okay, we already have `CPVRBuildins`. I will add "pvr.recordplayingchannel" there.

**Comment by ksooo** [comment]
Here we go with the PVR changes: Please feel fre to cherry-pick https://github.com/ksooo/xbmc/commit/03c6a949a2a32b52d39144c54c2ee55b7a65be6b

**Comment by ksooo** [comment]
@FernetMenta Updated commit: https://github.com/ksooo/xbmc/commit/899eb4f9b2fc77a1ea92e2b515163ff8c10ddf63

**Comment by ksooo** [comment]
@ronie Could you please annonounce the skinning engine changes introduced by this PR?

Infobools:
Replaced Player.CanRecord by PVR.CanRecordPlayingChannel (same semantics)
Replaced Player.Recording by PVR.IsRecordingPlayingChannel (same semantics)

Builtins:
Replaced PlayerControl(Record) by PVR.ToggleRecordPlayingChannel

I already did the required changes for Estuaray and Estouchy.

**Comment by notspiff** [comment]
if you do a post please include https://github.com/xbmc/xbmc/pull/13092

**Comment by ksooo** [comment]
@HitcherUK ^^^ heads-up for required Confluence skin changes. :-)

**Comment by HitcherUK** [comment]
> @HitcherUK ^^^ heads-up for required Confluence skin changes. :-)

Thanks.

**Pull request merged**